src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -208,28 +213,55 @@ private async Task<string> ComputeValueAsync(ComplexTargetFramework complexTarge
             return targetFrameworkAlias;
         }
 
-        private static bool IsNetCore5OrHigher(string targetFrameworkAlias)
+        private static bool IsNetCore5OrHigher(string? targetFrameworkAlias)
         {
-            return targetFrameworkAlias.Contains("5.0") || targetFrameworkAlias.Contains("6.0") || targetFrameworkAlias.Contains("7.0");
+            // Ideally, we want to use the TargetFrameworkIdentifier and TargetFrameworkVersion;
+            // however, in this case the target framework properties we have are for the currently set value,
+            // not the value we want to set it to (for example, if we go from netcoreapp3.1 to net5.0).
+            // The only property we have that describes the value to be set is the TargetFrameworkAlias which
+            // is passed to this method.
+
+            Match match = s_versionRegex.Match(targetFrameworkAlias);
```

Comment:
`Match` throws if passed a `null`.

https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.match?view=net-7.0#system-text-regularexpressions-regex-match(system-string)

We don't get a warning here because we target `net472` which isn't annotated.

When can the parameter be null?

Reply:
Good question. The only situation where it (TargetFrameworkAlias) can be null is if the TargetFramework is user-defined. However, I validate that before calling this method, so it should only be called when there's a value. I'll push some changes to reflect this. Thanks!